src/libs/ajax/file-browser-providers/AzureBlobStorageFileBrowserProvider.ts

```diff
@@ -185,6 +185,19 @@ const AzureBlobStorageFileBrowserProvider = ({
       return `azcopy copy '${blobUrl.href}' .`;
     },
     uploadFileToDirectory: async (directoryPath, file) => {
+      // This provider uses Azure Blob Storage's PUT blob API to upload files.
+      // This API supports blobs up to 5,000 MiB.
+      // https://learn.microsoft.com/en-us/rest/api/storageservices/put-blob#remarks
+      //
+      // If the user attempts to upload a larger file, we can fail fast with a more
+      // useful error message instead of waiting for the file to upload and the
+      // inevitable "Request body too large" error from Azure.
+      if (file.size > 5000 * 2 ** 20) {
```

Comment:
I like the shorthand of doing 2 ** 20, but can you extract it to a variable/constant with an intention revealing name?

Reply:
Done in https://github.com/DataBiosphere/terra-ui/pull/4516/commits/f5c6e9220b1863b0b8d140c90f68110d7b8c7c57.